### ISSUE TITLE
tracking issue - testing branch for CqlVector

### ISSUE THREAD

**Issue by jshook** [nosqlbench/nosqlbench#1325]
**NOTE**: the branch which has these changes is [vector-testing-1325](https://github.com/nosqlbench/nosqlbench/tree/vector-testing-1325)

While we wait for the PR for a more direct CqlVector value type to be reviewed and/or merged in the upstream CQL driver, we're going to be using our own branch for the purposes of testing. If the upstream doesn't merge our PR, we'll have to make a new type of our own and augment the driver codecs at runtime, if possible, to avoid the issues described below.

## Rationale

The initial CqlVector type in the java driver needs some fast-follow updates to make it performant for larger (typical) vector values. Since we need to be able to maintain roughly similar overhead between tests for empirical results and comparability, the existing type does not allow us to normalize a vector without some caller-side gymnastics and some extremely wasteful allocation patterns. Since a significant portion of a vector testing workload is actually vector variate generation (and associated normalization for an A-B scenario), the impact of this is high when the normalize operator is applied to the binding. This would likely make our comparable tests between simple scenarios like this unreliable due to a re-balancing of the combined testing apparatus.

The concerns above are not unique to nosqlbench, as it is, afterall, simply an application with respect to how the driver is used. The same API overhead and traversal patterns will likely affect others, particularly those who swim in ML and expect the vector type to be treated as a simple and idiomatic value type. We'll do what we can to improve this upstream as soon as we can. In the meantime, we'll work with a separate branch.

**Comment by msmygit** [comment]
I'm just going to write up the issue details here.

I’ve a workload yaml which has,
```
main-write:
    params:
      cl: TEMPLATE(write_cl,LOCAL_QUORUM)
    ops:
      ...
```
and
```
main-read:
    params:
      cl: TEMPLATE(read_cl,LOCAL_QUORUM)
    ops:
      ...
```
section respectively. In the calling script, if I use `read_cl=LOCAL_ONE`, it is mistakenly getting substituted for `main_cl` template.

Exception I receive is (notice, it’s applying it on the writes as opposed to just reads) this because Astra DB won’t support these write CLs:
```
Caused by: com.datastax.oss.driver.api.core.servererrors.InvalidQueryException: Provided value LOCAL_ONE is not allowed for Write Consistency Level (disallowed values are: [ANY, ONE, LOCAL_ONE])
```

I also attempted to directly hardcode the value within the `main-read` block as `cl: LOCAL_ONE` but it ends up calling the write. Changed the block names to `mainr` and `mainw` but still no luck. This all works fine if I use `cl: LOCAL_QUORUM` for **both** reads as well as writes. This is puzzling & is a current blocker.
